valohai_yaml/objs/input.py

```diff
@@ -52,3 +70,7 @@ def get_data(self) -> SerializedDict:
         else:
             data.pop("keep_directories", None)
         return data
+
+    def apply_edge_merge_mode(self) -> None:
+        if self.edge_merge_mode == EdgeMergeMode.REPLACE:
+            self.default = []
```

Comment:
I'm not sure this is a great idea. I'd expect the configuration object to be immutable.

Feels like the `Override.merge_with_step()` function should know about this instead.

Reply:
I have a strong opinion about it being included in Input as the property belongs to Input and should be applied 
before merging override with step. Mutating default based on property is not doing any damage as it's being serialized for node data right away which is required in Pipeline conversion.

If we still presist to do it in merge_with_step 
There we still have to mutate defaults before merge_dicts is called 

or if we do it in merge_dicts it becomes messy as that method is working for any type and this edge-merge-mode is specifically related to Input only.

Reply:
> as it's being serialized for node data right away

_here_, yes. 

If you have a strong opinion about it being included in this class, then I think it should at least be a function that doesn't internally mutate things (i.e. returns a shallow copy with `default` cleared):

```
input = input.with_edge_merge_mode_applied()
```